src/rust/ensogl/lib/components/src/card.rs

```diff
@@ -0,0 +1,71 @@
+//! This module provides the [`Card`] component.
+
+use crate::prelude::*;
+
+use crate::shadow;
+
+use ensogl_core::display;
+use ensogl_core::display::shape::*;
+use ensogl_core::display::object::{ObjectOps, Instance};
+use ensogl_theme as theme;
+
+
+
+// ============
+// === Card ===
+// ============
+
+/// The estimated margin that we need on all sides to include the shadow.
+const SHADOW_PX:f32 = 10.0;
+
+ensogl_core::define_shape_system! {
+    (style:Style,corner_radius:f32) {
+        let sprite_width  : Var<Pixels> = "input_size.x".into();
+        let sprite_height : Var<Pixels> = "input_size.y".into();
+
+        let width  = sprite_width - SHADOW_PX.px() * 2.0;
+        let height = sprite_height - SHADOW_PX.px() * 2.0;
+        let color  = style.get_color(theme::graph_editor::node::background);
+        let rect   = Rect((&width,&height)).corners_radius(corner_radius);
+        let shape  = rect.fill(color);
+
+        let shadow = shadow::from_shape(rect.into(),style);
+
+        (shadow + shape).into()
+    }
+}
+
+/// A simple display object consisting of a white rectangle with rounded corners and a shadow. This
+/// can be used as a background for other components.
+#[derive(Debug,Clone,CloneRef)]
+pub struct Card(View);
+
+impl Card {
+    /// Create a new card.
+    pub fn new() -> Self {
+        Card(View::new(Logger::new("Card")))
+    }
+
+    /// Set the size of the card's base shape. The shadow can reach beyound this.
+    pub fn resize(&self,size:Vector2) {
+        let shadow_margin = Vector2(SHADOW_PX,SHADOW_PX);
+        self.0.size.set(size+2.0*shadow_margin);
+    }
```

Comment:
I object to 1. All other components use `resize`. If we rename it here then we should rename it everywhere consistently. But I also disagree that `set_size` is a better name. `resize` makes it clearer that this has an actual effect on screen, rather than just setting some internal attribute, in my opinion.

Reply:
I don't agree with that TBH. Naming consistency should be the most basic guideline for us. If all such "setters" start with `set_` prefix, then not following this rule with one function seems like a dirty naming design to me. And yes, if other components use it the same way, it should be changed there as well. Especially when included in FRP API, as we use `set_` prefix there for all setters.